support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -16,46 +16,87 @@ limitations under the License.
 package controllers
 
 import (
+	"bytes"
 	"context"
+	"errors"
+	"strings"
+	"text/template"
 
 	"github.com/sirupsen/logrus"
 	federation "github.com/spiffe/spire/support/k8s/k8s-workload-registrar/federation"
 	spiffeidv1beta1 "github.com/spiffe/spire/support/k8s/k8s-workload-registrar/mode-crd/api/spiffeid/v1beta1"
 
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	PodNameIDLabel           string = "pod_name"
+	PodUIDLabel              string = "pod_uid"
+	NamespaceIDLabel         string = "namespace"
+	PodServiceAccountIDLabel string = "service_account"
+	PodHostnameLabel         string = "hostname"
+	PodNodeNameLabel         string = "node_name"
+)
+
 // PodReconcilerConfig holds the config passed in when creating the reconciler
 type PodReconcilerConfig struct {
-	Client             client.Client
-	Cluster            string
-	Ctx                context.Context
-	DisabledNamespaces []string
-	Log                logrus.FieldLogger
-	PodLabel           string
-	PodAnnotation      string
-	Scheme             *runtime.Scheme
-	TrustDomain        string
+	Client                client.Client
+	Cluster               string
+	Ctx                   context.Context
+	DisabledNamespaces    []string
+	Log                   logrus.FieldLogger
+	PodLabel              string
+	PodAnnotation         string
+	Scheme                *runtime.Scheme
+	TrustDomain           string
+	IdentityTemplate      string
+	Context               map[string]string
+	IdentityTemplateLabel string
+}
+
+// PodInfo is created for every processed Pod and it holds pod specific information
+type PodInfo struct {
+	PodServiceAccountIDLabel string
+	NamespaceIDLabel         string
+	PodNameIDLabel           string
+	PodUIDLabel              string
+	PodHostnameLabel         string
+	PodNodeNameLabel         string
+}
+
+// IdentityMaps is used for forming the text from the templates
+type IdentityMaps struct {
+	Context map[string]string
+	Pod     PodInfo
 }
 
 // PodReconciler holds the runtime configuration and state of this controller
 type PodReconciler struct {
 	client.Client
-	c PodReconcilerConfig
+	c             PodReconcilerConfig
+	identityTempl *template.Template
 }
 
 // NewPodReconciler creates a new PodReconciler object
-func NewPodReconciler(config PodReconcilerConfig) *PodReconciler {
-	return &PodReconciler{
-		Client: config.Client,
-		c:      config,
+func NewPodReconciler(config PodReconcilerConfig) (*PodReconciler, error) {
+
+	tpl := config.IdentityTemplate
```

Comment:
Can we just directly use `config.IdentityTemplate` instead of having the extra `tpl` variable.